extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java

```diff
@@ -24,25 +28,51 @@ public void apicurioRegistryAvro(BuildProducer<ReflectiveClassBuildItem> reflect
                 "io.apicurio.registry.serde.avro.AvroKafkaDeserializer",
                 "io.apicurio.registry.serde.avro.AvroKafkaSerializer"));
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
                 "io.apicurio.registry.serde.strategy.SimpleTopicIdStrategy",
                 "io.apicurio.registry.serde.strategy.TopicIdStrategy",
                 "io.apicurio.registry.serde.avro.DefaultAvroDatumProvider",
                 "io.apicurio.registry.serde.avro.ReflectAvroDatumProvider",
                 "io.apicurio.registry.serde.avro.strategy.RecordIdStrategy",
                 "io.apicurio.registry.serde.avro.strategy.TopicRecordIdStrategy"));
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
                 "io.apicurio.registry.serde.DefaultSchemaResolver",
                 "io.apicurio.registry.serde.DefaultIdHandler",
                 "io.apicurio.registry.serde.Legacy4ByteIdHandler",
                 "io.apicurio.registry.serde.fallback.DefaultFallbackArtifactProvider",
                 "io.apicurio.registry.serde.headers.DefaultHeadersHandler"));
+
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
+                "io.apicurio.rest.client.auth.exception.NotAuthorizedException",
+                "io.apicurio.rest.client.auth.exception.ForbiddenException",
+                "io.apicurio.rest.client.auth.exception.AuthException",
+                "io.apicurio.rest.client.auth.exception.AuthErrorHandler",
+                "io.apicurio.rest.client.auth.request.TokenRequestsProvider",
+                "io.apicurio.rest.client.request.Request",
+                "io.apicurio.rest.client.auth.AccessTokenResponse",
+                "io.apicurio.rest.client.auth.Auth",
+                "io.apicurio.rest.client.auth.BasicAuth",
+                "io.apicurio.rest.client.auth.OidcAuth"));
+    }
+
+    @BuildStep
+    void registerSPIClient(BuildProducer<ServiceProviderBuildItem> services) throws IOException {
+
+        services.produce(
+                new ServiceProviderBuildItem(ApicurioHttpClientProvider.class.getName(),
+                        "io.apicurio.rest.client.JdkHttpClientProvider"));
```

Comment:
That's not the one we want. We want the Vert.x Web Client.

Reply:
Yes, this probably explains the problems you're having with Vert.x connections @secondsun

Reply:
When I set that to the vertx client I get this error instead 
```
2021-12-01 17:05:32,501 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.util.ServiceConfigurationError: io.apicurio.rest.client.spi.ApicurioHttpClientProvider: Provider io.apicurio.rest.client.JdkHttpClientProvider not found
	at java.util.ServiceLoader.fail(ServiceLoader.java:589)
	at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1212)
	at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:106)
	at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.util.ServiceLoader$2.next(ServiceLoader.java:1306)
	at java.util.ServiceLoader$2.next(ServiceLoader.java:1297)
	at java.util.ServiceLoader$3.next(ServiceLoader.java:1395)
	at io.apicurio.rest.client.auth.OidcAuth.resolveProviderInstance(OidcAuth.java:81)
	at io.apicurio.rest.client.auth.OidcAuth.resolveApicurioHttpClient(OidcAuth.java:85)
	at io.apicurio.rest.client.auth.OidcAuth.<init>(OidcAuth.java:65)
	at io.apicurio.registry.serde.AbstractSchemaResolver.configureAuthWithUrl(AbstractSchemaResolver.java:232)
	at io.apicurio.registry.serde.AbstractSchemaResolver.configureClientWithBearerAuthentication(AbstractSchemaResolver.java:203)
	at io.apicurio.registry.serde.AbstractSchemaResolver.configure(AbstractSchemaResolver.java:75)
	at io.apicurio.registry.serde.DefaultSchemaResolver.configure(DefaultSchemaResolver.java:56)
	at io.apicurio.registry.serde.SchemaResolverConfigurer.configure(SchemaResolverConfigurer.java:75)
	at io.apicurio.registry.serde.AbstractKafkaSerDe.configure(AbstractKafkaSerDe.java:68)
	at io.apicurio.registry.serde.AbstractKafkaDeserializer.configure(AbstractKafkaDeserializer.java:62)
	at io.apicurio.registry.serde.avro.AvroKafkaDeserializer.configure(AvroKafkaDeserializer.java:68)
	at io.smallrye.reactive.messaging.kafka.fault.DeserializerWrapper.configure(DeserializerWrapper.java:82)
	at io.smallrye.reactive.messaging.kafka.impl.ReactiveKafkaConsumer.<init>(ReactiveKafkaConsumer.java:79)
	at io.smallrye.reactive.messaging.kafka.impl.KafkaSource.<init>(KafkaSource.java:93)
	at io.smallrye.reactive.messaging.kafka.KafkaConnector.getPublisherBuilder(KafkaConnector.java:173)
	at io.smallrye.reactive.messaging.kafka.KafkaConnector_ClientProxy.getPublisherBuilder(KafkaConnector_ClientProxy.zig:159)
	at io.smallrye.reactive.messaging.impl.ConnectorFactories.lambda$wrap$4(ConnectorFactories.java:80)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.createPublisher(ConfiguredChannelFactory.java:174)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.register(ConfiguredChannelFactory.java:136)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.initialize(ConfiguredChannelFactory.java:108)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory_ClientProxy.initialize(ConfiguredChannelFactory_ClientProxy.zig:189)
	at java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at io.smallrye.reactive.messaging.extension.MediatorManager.start(MediatorManager.java:189)
	at io.smallrye.reactive.messaging.extension.MediatorManager_ClientProxy.start(MediatorManager_ClientProxy.zig:220)
	at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:41)
	at 
```


I think this is because the SPI declaration for the jdk client is in the jdk client jar https://github.com/Apicurio/apicurio-common-rest-client/tree/main/rest-client-jdk/src/main/resources/META-INF/services, but I haven't tested this assumption yet